tests/test_core.py

```diff
@@ -381,6 +381,20 @@ def test_method_incorrect_argument_count(self):
         with self.assertRaises(TypeError):
             obj.mutateIntFieldWithValue_(123, "extra argument")
 
+    def test_method_incorrect_argument_type(self):
+        """Attempting to call a method with the wrong type of argument throws an exception."""
+
+        Example = ObjCClass("Example")
+        obj = Example.alloc().init()
+
+        with self.assertRaisesRegex(
+            TypeError,
+            # This pattern match needs to be flexible because the exact
+            # text returned in the ctype error changes between Python versions.
+            r"argument 3: .*?: wrong type \(mutateIntFieldWithValue: argtypes: c_int\)",
```

Comment:
Can we list the possible texts using the `(A|B)` syntax, rather than allowing just anything here?

Reply:
Why does it say "argument 3"? I could understand argument 1 being `self`, or the Objective-C equivalent, but what's argument 2?

Reply:
Argument 1 is indeed `self`. Argument 2 is `_cmd`, the selector (method name) of the method being called. Argument 3 and on are the actual method arguments.

Reply:
I've taken an alternate approach - include both messages as exact text, with a "if sys.version" check; that way, when we (eventually) deprecate Python 3.9, pyupgrade will automatically do the cleanup and remove the dead branch.